junit-platform-commons/src/main/java/org/junit/platform/commons/support/conversion/Converter.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.support.conversion;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import org.apiguardian.api.API;
+
+/**
+ * {@code Converter} is an abstraction that allows an input object to
+ * be converted to an instance of a different class.
+ *
+ * <p>Implementations are loaded via the {@link java.util.ServiceLoader} and must
+ * follow the service provider requirements. They should not make any assumptions
+ * regarding when they are instantiated or how often they are called. Since
+ * instances may potentially be cached and called from different threads, they
+ * should be thread-safe.
+ *
+ * <p>Extend {@link TypedConverter} if your implementation always converts
+ * from a given source type into a given target type and does not need access to
+ * the {@link ClassLoader} to perform the conversion.
+ *
+ * @since 1.13
+ * @see ConversionSupport
+ * @see TypedConverter
+ */
+@API(status = EXPERIMENTAL, since = "1.13")
+public interface Converter {
+
+	/**
+	 * Determine if the supplied source object can be converted into an instance
+	 * of the specified target type.
+	 *
+	 * @param source the source object to convert; may be {@code null} but only
+	 * if the target type is a reference type
+	 * @param targetType the descriptor of the type the source should be converted into;
+	 * never {@code null}
+	 * @return {@code true} if the supplied source can be converted
+	 */
+	boolean canConvert(Object source, TypeDescriptor targetType);
```

Comment:
Initially, this was:

```suggestion
	boolean canConvert(Object source, TypeDescriptor targetType, ClassLoader classLoader);
```

However, I couldn't see the benefit of the `ClassLoader` parameter here, so I deleted it.

Plus, I'm not entirely sure if this should rather be:

```suggestion
	boolean canConvert(TypeDescriptor sourceType, TypeDescriptor targetType);
```

The way `Converter` is currently integrated into `ConversionSupport` makes it easy to provide the source instance, but I wonder if this would be flexible enough with a bit of foresight.

Reply:
I think we should pass an _additional` `TypeDescriptor sourceType` to both `canConvert` and `convert`.

Reply:
Done in 5190e37.